plugins/participants.js

```diff
@@ -0,0 +1,167 @@
+module.exports = (controller) => {
+    const renderParticipantsBlocks = async (participants) => {
+        return Promise.all(
+            participants.map((participant) =>
+                renderParticipantBlocks(participant)
+            )
+        ).then((participants) => participants.flat());
+    };
+
+    const renderParticipantBlocks = async (participant) => {
+        const links = participant.links.map((link) => {
+            return {
+                type: "section",
+                text: {
+                    type: "mrkdwn",
+                    text: `:link: <${link.url}|${link.name}>`,
+                },
+            };
+        });
+        const gauges = participant.gauges.map((gauge) => {
+            return {
+                type: "section",
+                text: {
+                    type: "mrkdwn",
+                    text: participants.getGaugeString(gauge),
+                },
+            };
+        });
+        const details = participant.details.map((detail) => {
+            return {
+                type: "mrkdwn",
+                text: `*${detail.name}*: ${detail.value}`,
+            };
+        });
+        return [
+            {
+                type: "section",
+                text: {
+                    type: "mrkdwn",
+                    text: `:bust_in_silhouette: *${participant.info.real_name}*`,
+                },
+                accessory: {
+                    type: "image",
+                    image_url: participant.info.profile.image_72,
+                    alt_text: participant.info.real_name,
+                },
+                fields: details.length ? details : undefined,
+            },
+            ...gauges,
+            ...links,
+        ];
+    };
+
+    const renderNoContentBlocks = () => {
```

Comment:
can these helper functions `renderNoContentBlocks`, `renderParticipantBlocks`, and `renderParticipantsBlocks` be moved out of this scope?  it doesn't look like they need to be here, and it makes the code a bit confusing to read, at least for me.

Reply:
renderParticipantBlocks calls participantsPlugin.getGuageString() so it needs to be in the same scope as participantsPlugin (i renamed the object to make it more clear). I would rather keep all the methods in the same scope.